src/plugins/embeddable/public/components/quick_edit_flyout/index.tsx

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import { render } from 'react-dom';
+import { take } from 'rxjs/operators';
+import { ApplicationStart } from 'kibana/public';
+import {
+  EuiIcon,
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFlyoutHeader,
+  EuiFlyoutFooter,
+  EuiTitle,
+  EuiLink,
+} from '@elastic/eui';
+import {
+  Container,
+  EmbeddableInput,
+  EmbeddableOutput,
+  SavedObjectEmbeddableInput,
+  IEmbeddable,
+  EmbeddableEditorState,
+  EmbeddableStateTransfer,
+} from '../..';
+
+interface Props {
+  onClose: () => void;
+  onCancel: () => void;
+  embeddable: IEmbeddable<EmbeddableInput, EmbeddableOutput>;
+  application: ApplicationStart;
+  stateTransferService?: EmbeddableStateTransfer;
+}
+
+interface NavigationContext {
+  app: string;
+  path: string;
+  state?: EmbeddableEditorState;
+}
+
+export class QuickEditFlyout extends React.Component<Props> {
+  private currentAppId: string | undefined;
+  private onSave: () => void;
+
+  constructor(props: Props) {
+    super(props);
+
+    // temporary no-op
+    this.onSave = () => {};
+
+    const { embeddable, application } = this.props;
+    if (embeddable.getQuickEditControl) {
+      embeddable.getQuickEditControl()?.then(({ control, onSave }) => {
+        if (control) {
+          render(control, document.getElementById('hello-put-it-here'));
+        }
+
+        this.onSave = onSave;
+      });
+    }
+
+    if (application?.currentAppId$) {
+      application.currentAppId$
+        .pipe(take(1))
+        .subscribe((appId: string | undefined) => (this.currentAppId = appId));
+    }
+  }
+
+  private async goToApp(context: IEmbeddable) {
```

Comment:
This is a copy of the logic that lives in the "edit" UI action to go to the editor the embeddable was created in. We can probably refactor this into a shared library to avoid code duplication

Reply:
Also, in this function,  we need to restore the default state of the embeddable